influence_benchmark/stats/retroactive_evals.py

```diff
@@ -267,40 +264,47 @@ def sync_evaluate_iteration(self, all_transcripts_with_env, total_transcripts):
                 if result == "DONE":
                     completed_processes += 1
                 else:
-                    results.append(result)
+                    results.extend(result)
                     pbar.update(len(result))
 
             for p in processes:
                 p.join()
 
-        flat_results = list(itertools.chain(*results))
-        return flat_results
+        return results
 
-    def _process_batches(self, batches_tuple, device, results_queue):
+    def _process_chunk(self, chunk, device, results_queue):
         backend = self.backend_class(
             model_name=self.backend_config["model_name"],
             model_id=self.backend_config["model_id"],
             lora_path=self.backend_config["lora_path"],
             device=device,
-        )  # type: ignore
-        for batch_tuple in batches_tuple:
-            start, batch = batch_tuple
-            batch_results = self.evaluate_batch(batch, start, backend)
-            results_queue.put(batch_results)
+        )
+        vectorized_assessors = self.vectorized_assessors_for_backend(backend)
+        batch_size = self.per_device_batch_size
+        num_batches = (len(chunk) + batch_size - 1) // batch_size  # Ceiling division
```

Comment:
Instead of computing the number of batches, you can just use indexing directly in the for loop, and when you get a smaller batch size you know you're done?